src/index.spec.ts

```diff
@@ -1,777 +1,898 @@
+
+            let src = path.join(options.rootDir, 'manifest');
+            let dest = path.join(stagingFolderPath, 'manifest');
+            fsExtra.copySync(src, dest);
+        });
+
+        it('should return our error if signingPassword is not supplied', async () => {
+            options.signingPassword = undefined;
+            try {
+                await rokuDeploy.signExistingPackage(options);
+                assert.fail('Exception should have been thrown');
+            } catch (e) {
+                expect(e.message).to.equal('Must supply signingPassword');
             }
-        ];
-        await prepublishToStaging(options);
-        expect(file('out/.roku-deploy-staging/source/main.brs')).to.exist;
-    });
+        });
 
-    it('handles new src;dest style', async () => {
-        options.files = [
-            {
-                src: 'manifest',
-                dest: ''
-            },
-            {
-                src: 'source/**/*',
-                dest: 'source/'
-            },
-            {
-                src: 'source/main.brs',
-                dest: 'source/main.brs'
+        it('should return an error if there is a problem with the network request', async () => {
+            let error = new Error('Network Error');
+            try {
+                //intercept the post requests
+                sinon.stub(rokuDeploy.__request, 'post').callsFake((_, callback) => process.nextTick(callback, error));
+                await rokuDeploy.signExistingPackage(options);
+                assert.fail('Exception should have been thrown');
+            } catch (e) {
+                expect(e).to.equal(error);
             }
-        ];
-        await prepublishToStaging(options);
-        expect(file('out/.roku-deploy-staging/manifest')).to.exist;
-        expect(file('out/.roku-deploy-staging/source/main.brs')).to.exist;
-    });
+        });
 
-    it('handles renaming files', async () => {
-        options.files = [
-            {
-                src: 'manifest',
-                dest: ''
-            },
-            {
-                src: 'source/main.brs',
-                dest: 'source/renamed.brs'
+        it('should return our error if it received invalid data', async () => {
+            try {
+                //intercept the post requests
+                sinon.stub(rokuDeploy.__request, 'post').callsFake((_, callback) => process.nextTick(callback));
+                await rokuDeploy.signExistingPackage(options);
+                assert.fail('Exception should have been thrown');
+            } catch (e) {
+                expect(e.message).to.equal('Invalid response');
             }
-        ];
-        await prepublishToStaging(options);
-        expect(file('out/.roku-deploy-staging/source/renamed.brs')).to.exist;
-    });
+        });
+
+        it('should return an error if failure returned in response', async () => {
+            let body = `   <div style="display:none">
```

Comment:
Format this html a little nicer, or add a comment why it needs to be exactly this format.

```typescript
let body = `   
    <div style="display:none">
        <font color="red">Failed: Invalid Password.</font>
   </div>
`;

Reply:
I'm open to thoughts on this. I copied it exactly how it was on the device. It looks ugly in vs code since the spacing is off. I'll do a comment for now but am open to switching but didn't want the regex check to not be exactly what was happening.